speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp

```diff
@@ -29,43 +38,226 @@ TEST(MediatorTests, GetDistributedMediatorTemplated)
+{
+    SDAConfigMediator::ClearInstance();
+    ASSERT_TRUE(SetupSingletonsFolder());
+    tParticipantControl arr = {NULL, NULL, NULL, p_bool1, p_bool2, p_bool3};
+    SDAConfigMediator::GetInstance()->SetPControlSettings(arr);
+    const SDAConfig config = SDAConfigMediator::GetInstance()->GetDecisionMaker()->Config;
+    tParticipantControl pControl = config.GetPControlSettings();
+    ASSERT_EQ(arr.ForceFeedback, pControl.ForceFeedback);
+    ASSERT_EQ(arr.RecordSession, pControl.RecordSession);
+    ASSERT_EQ(arr.BBRecordSession, pControl.BBRecordSession);
+}
+
+/// @brief Tests the Mediator ParticipantControlSettings for every possible boolean combination (last 3)
+BEGIN_TEST_COMBINATORIAL(MediatorTests, PControlSettings2)
+bool booleans[] = {false, true};
+END_TEST_COMBINATORIAL3(PControlTest2Mediator, booleans, 2, booleans, 2, booleans, 2)
+
+/// @brief Tests if the Mediator sets and gets the MaxTime correctly
+TEST(MediatorTests, MaxTimeTest)
+{
+    Random random;
+    for (int i = 0; i < 20; i++)
+    {
+        SDAConfigMediator::ClearInstance();
+        ASSERT_TRUE(SetupSingletonsFolder());
+        int maxTime = random.NextInt();
+        SDAConfigMediator::GetInstance()->SetMaxTime(maxTime);
+        const SDAConfig config = SDAConfigMediator::GetInstance()->GetDecisionMaker()->Config;
+        ASSERT_EQ(maxTime, config.GetMaxTime());
+    }
+}
+
+/// @brief Tests if the Mediator sets and gets the UserID correctly
+TEST(MediatorTests, UserIDTest)
+{
+    Random random;
+    char buf[32];
+    for (int i = 0; i < 20; i++)
+    {
+        SDAConfigMediator::ClearInstance();
+        ASSERT_TRUE(SetupSingletonsFolder());
+        int userID = random.NextInt();
+        sprintf(buf, "%d", userID);
+        SDAConfigMediator::GetInstance()->SetUserId(buf);
+        const SDAConfig config = SDAConfigMediator::GetInstance()->GetDecisionMaker()->Config;
+        ASSERT_EQ(buf, config.GetUserId());
+    }
+}
+
+/// @brief Tests if the Mediator sets and gets the BlackBoxFilePath correctly
+TEST(MediatorTests, BlackBoxFilePathTest)
+{
+    Random random;
+    for (int j = 0; j <= 10; j++)
+    {
+        SDAConfigMediator::ClearInstance();
+        ASSERT_TRUE(SetupSingletonsFolder());
+        char path[256];
+        int length = 2 + 254 * j / 10;
+        for (int i = 0; i < length - 1; i++)
+        {
+            char c;
+            do
+            {  // Make sure no character is assigned the null character
+                c = (char)random.NextByte();
+            } while (c == '\0');
+            path[i] = c;
+        }
+        path[length - 1] = '\0';
+        SDAConfigMediator::GetInstance()->SetBlackBoxFilePath(path);
+        const SDAConfig config = SDAConfigMediator::GetInstance()->GetDecisionMaker()->Config;
+        const char* configPath = config.GetBlackBoxFilePath();
+        for (int i = 0; i < length; i++)
+        {
+            ASSERT_EQ(path[i], configPath[i]);
```

Comment:
Can ASSERT_STREQ be used instead? If not, ensure to check the length of both strings as well: if configPath is longer than length, this is not caught by the test.